### PR TITLE
Migrate from deprecated rspec-system-puppet

### PR DESCRIPTION
**Pull request by jimcroft**
Am keen to see https://github.com/puppetlabs/puppetlabs-splunk/pull/9 and other improvements in this module. As a starter I thought it made sense to update the testing framework used as puppetlabs/rspec-system-puppet is retired in favour of beaker-rspec.

Happy to make any changes needed to get this merged in.

### CODE REVIEW

**Comment by tphoney** [comment]
Thanks for the great work you have done, could you please squash your commits and we will get this merged.

**Comment by jimcroft** [comment]
Thanks @tphoney will do. 

I've got a couple of tweaks still to make but will then squash and let you know the PR is ready for review/merge. Quick question, I was going to add Gemfile.lock to the repo to pin gems to versions I know work for the test suite. Any issue with that?

**Comment by tphoney** [comment]
Normally the gems are pinned in the Gemfile(loads of examples in other puppet modules), like this: 
source 'https://rubygems.org'

puppetversion = ENV.key?('PUPPET_VERSION') ? "= #{ENV['PUPPET_VERSION']}" : ['>= 3.3']
gem 'puppet', puppetversion
gem 'puppetlabs_spec_helper', '>= 0.1.0'
gem 'puppet-lint', '>= 0.3.2'
gem 'facter', '>= 1.7.0'

The Gemfile.lock should not be checked in.

**Comment by jimcroft** [comment]
:+1:

**Comment by jimcroft** [comment]
@tphoney sorry for the delay, think this is good to merge now. assuming we're I'll open a separate PR containing the cluster role changes by @hyclak and appropriate tests.

**Comment by hyclak** [comment]
Thanks for taking this on @jimcroft. I'm no longer working on Splunk internally, so I don't have any cycles for development anymore.

**Comment by tphoney** [comment]
It looks like this needs to be rebased. Sorry that this has went a little stale.

**Comment by hunner** [comment]
Ticketed in https://tickets.puppetlabs.com/browse/MODULES-3112 :+1:

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 *.swp
+files/splunk
+files/universalforwarder
```
Comment (dhollinger) [inline .gitignore]: The `.gitignore` file is managed by modulesync. If these files still need to be added to the `.gitignore` they should be added to the .sync.yml file.

**Comment by dhollinger** [comment]
From what I can tell, these changes are no longer needed as many of the issues this PR was trying to solve have been taken care of in other merges.

The current version of puppet-splunk and the puppet3 legacy branch both have accounted for all of the changes listed, other than the addition to the README file and the addition of some "legacy mode" logic. 

Rather than try and rebase this onto the existing, more current codebase, I would suggest we close the PR and review if the remaining changes are even necessary anymore.

**Comment by tphoney** [comment]
based on @dhollinger feedback, looks like it can be closed.

**Pull request closed without merging**